source/slang/slang-ir-link.cpp

```diff
@@ -1534,7 +1532,7 @@ void convertAllAtomicUintToStructuredBufferUsages(
         uint32_t maxOffset = 0;
         for (auto& i : bindingToInstList.second)
         {
-            int currOffset = i->findDecoration<IRGLSLOffsetDecoration>()->getOffset()->getValue();
+            uint32_t currOffset = uint32_t(i->findDecoration<IRGLSLOffsetDecoration>()->getOffset()->getValue());
```

Comment:
I think maxOffset and currOffset should be 64bit.

Reply:
👍 32bit is a mistake, vkdevice limits allow more than 4GB worth of data